app/components/amount-input-screen/number-pad-reducer.ts

```diff
@@ -0,0 +1,156 @@
+import { WalletOrDisplayCurrency } from "@app/types/amounts"
+
+export type NumberPadNumber = {
+  majorAmount: string
+  minorAmount: string
+  hasDecimal: boolean
+}
+
+export type NumberPadReducerState = {
+  numberPadNumber: NumberPadNumber
+  numberOfDecimalsAllowed: number
+  currency: WalletOrDisplayCurrency
+}
+
+export const NumberPadReducerActionType = {
+  SetAmount: "SetAmount",
+  HandleKeyPress: "HandleKeyPress",
+  ClearAmount: "ClearAmount",
+} as const
+
+export type NumberPadReducerAction =
+  | {
+      action: typeof NumberPadReducerActionType.SetAmount
+      payload: {
+        numberPadNumber: NumberPadNumber
+        numberOfDecimalsAllowed: number
+        currency: WalletOrDisplayCurrency
+      }
+    }
+  | {
+      action: typeof NumberPadReducerActionType.HandleKeyPress
+      payload: {
+        key: Key
+      }
+    }
+  | {
+      action: typeof NumberPadReducerActionType.ClearAmount
+    }
+
+export const Key = {
+  Backspace: "⌫",
+  0: "0",
+  1: "1",
+  2: "2",
+  3: "3",
+  4: "4",
+  5: "5",
+  6: "6",
+  7: "7",
+  8: "8",
+  9: "9",
+  Decimal: ".",
```

Comment:
in Europe and probably other locale, this would be a `,` (I think). just fyi, I think we should focus on those i18n details later